test/api/junction_detection_methods_test.cpp

```diff
@@ -0,0 +1,184 @@
+//                         Chromosome: chr21
+std::string expected_res_cigar
+{
+    "Reference\tchr21\t41972616\tForward\tRead\t0\t2294\tForward\t1\n"
+    "Reference\tchr21\t41972616\tReverse\tRead\t0\t3975\tReverse\t1\n"
+};
+
+// Reference\tchr22\t17458417\tForward\tReference\tchr21\t41972615\tForward\tm41327/11677/CCS
+// BND from SA Tag - Sequence Type: Reference; Chromosome: chr22; Position: 17458417; Orientation: Forward
+//                   Sequence Type: Reference; Chromosome: chr21; Position: 41972615; Orientation: Forward
+//                   Sequence Name: m41327/11677/CCS
+std::string expected_res_split
+{
+    "Reference\tchr22\t17458417\tForward\tReference\tchr21\t41972615\tForward\t1\n"
+    "Reference\tchr22\t17458418\tForward\tReference\tchr21\t41972616\tForward\t2\n"
+};
+
+std::string expected_res_pair = "";
+std::string expected_res_depth = "";
+
+void check_output_and_cleanup(std::string expected_res)
+{
+    std::string std_cout = testing::internal::GetCapturedStdout();
+    EXPECT_EQ(expected_res, std_cout);
+
+    // cleanup
+    std::filesystem::remove(tmp_dir/"detect_breakends_out_short.fasta");        // remove output
+
+    return;
+}
+
+/* -------- detection methods tests -------- */
+/* -------- single methods -------- */
+
+TEST(junction_detection, single_method_only)
+{
+    for (uint8_t method : all_methods)
+    {
+        std::filesystem::remove(tmp_dir/"detect_breakends_out_short.fasta");    // remove old output if existent
+
+        testing::internal::CaptureStdout();
+        seqan3::debug_stream << "-----------------------------------------------------------------------\n"
+                             << "Test Method: " << method << '\n'
+                             << "-----------------------------------------------------------------------\n";
+        detect_junctions_in_alignment_file(DATADIR"simulated.minimap2.hg19.coordsorted_cutoff.sam",
+                                        tmp_dir/"detect_breakends_out_short.fasta",
+                                        {method},
+                                        simple_clustering,
+                                        no_refinement,
+                                        sv_default_length);
+
+        if (method == 1)
+        {
+            check_output_and_cleanup(expected_res_cigar);
+        }
+        else if (method == 2)
+        {
+            check_output_and_cleanup(expected_res_split);
+        }
+        else if (method == 3)
+        {
+            check_output_and_cleanup(expected_res_pair);
+        }
+        else // (method == 4)
+        {
+            check_output_and_cleanup(expected_res_depth);
+        }
+    }
+}
+
+/* -------- method pairs -------- */
+
+TEST(junction_detection, method_pairs)
+{
+    for (uint8_t method_i : all_methods)
+    {
+        for (uint8_t method_j = method_i + 1; method_j <= all_methods.size(); ++method_j)
+        {
+            if (method_i != method_j)
```

Comment:
I don't think this does what you want.
By accident `all_methods.size()  == 4` and `all_methods.back() == 4`.
But if you used `all_methods = {4, 5, 6 ,7}` The code wouldn't do what you wanted.

Suggestion:
```suggestion
    for (uint8_t method_i : all_methods)
    {
        for (uint8_t method_j : all_methods)
        {
            if (method_i >= method_j) continue; // only check in one order to safe time
```